sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -0,0 +1,389 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import fs from "fs-extra";
+import nise from "nise";
+import queryString from "query-string";
+import {
+  isBrowser,
+  blobToString,
+  escapeRegExp,
+  env,
+  TestInfo,
+  isPlayingBack,
+  isRecording
+} from "./utils";
+import { customConsoleLog } from "./customConsoleLog";
+
+let nock: any;
+
+let replaceableVariables: { [x: string]: string } = {};
+export function setReplaceableVariables(a: { [x: string]: string }): void {
+  replaceableVariables = a;
+  if (isPlayingBack()) {
+    // Providing dummy values to avoid the error
+    Object.keys(a).map((k) => {
+      env[k] = a[k];
+    });
+  }
+}
+
+let replacements: any[] = [];
+export function setReplacements(maps: any): void {
+  replacements = maps;
+}
+
+export function setEnviromentOnLoad() {
+  if (!isBrowser() && (isRecording || isPlayingBack)) {
+    nock = require("nock");
+  }
+
+  if (isBrowser() && isRecording()) {
+    customConsoleLog();
+  }
+
+  if (isPlayingBack()) {
+    // Providing dummy values to avoid the error [ENVs for storage packages]
+    env.ACCOUNT_NAME = "fakestorageaccount";
+    env.ACCOUNT_KEY = "aaaaa";
+    env.ACCOUNT_SAS = "aaaaa";
+    env.STORAGE_CONNECTION_STRING = `DefaultEndpointsProtocol=https;AccountName=${
+      env.ACCOUNT_NAME
+    };AccountKey=${env.ACCOUNT_KEY};EndpointSuffix=core.windows.net`;
+  }
+}
+
+const skip = [
+  // Abort
+  "browsers/aborter/recording_should_abort_after_aborter_timeout.json"
+];
+
+export abstract class BaseRecorder {
+  protected readonly filepath: string;
+  public uniqueTestInfo: TestInfo = { uniqueName: {}, newDate: {} };
+
+  constructor(runtime: string, testHierarchy: string, testTitle: string, ext: string) {
```

Comment:
`runtime` -> `platform`

Consider having the type has either an enum or `"node" | "browser"`

Reply:
Changed `testHierarchy` ->  `testSuiteTitle`, `runtime` -> `platform`, 
Union type `platform: "node" | "browsers"` and removed `ext` argument.